File: src/PainelProjetosPVT.Api/Dominio/Entidades/Apontamento.cs
using PainelProjetosPVT.Api.Dominio.Enums;

namespace PainelProjetosPVT.Api.Dominio.Entidades;

public class Apontamento
{
    public int Id { get; set; }

    public int ProjetoId { get; set; }

    public Projeto Projeto { get; set; } = null!;

    public int AnalistaId { get; set; }

    public Analista Analista { get; set; } = null!;

    public DateOnly Data { get; set; }

    public decimal QuantidadeHoras { get; set; }

    public string? Descricao { get; set; }

    public StatusApontamento Status { get; set; }
}

File: frontend/src/types/apontamento.ts
export type StatusApontamento =
  | "Pendente"
  | "Aprovado"
  | "Rejeitado";

export interface Apontamento {
  id: number;
  projetoId: number;
  analistaId: number;
  data: string;
  quantidadeHoras: number;
  descricao: string;
  status: StatusApontamento;
}

export interface CriarApontamento {
  analistaId: number;
  data: string;
  quantidadeHoras: number;
  descricao: string;
}